### ISSUE TITLE
TypeError: Cannot read property 'replace' of undefined

### ISSUE THREAD

**Issue by tcury** [gadelkareem/sails-dynamodb#7]
I'm using sails 0.11.0 and sails-dynamodb 0.12.0.

I get this error:
`TypeError: Cannot read property 'replace' of undefined`

Comment out the line 847 of index.js I have more this error:
`error: { [Error: the value of id is not allowed to be undefined]`

**Comment by devinivy** [comment]
What causes that error to appear?  Is there a stack trace?

**Comment by tcury** [comment]
When creating a new table, I think. Because I'm using passport.js (sails-generate-auth) and I try to add a new user. Bellow the stack trace:
`
c:\Users\tiago.cury\doceasy\node_modules\sails-dynamodb\index.js, create error: { [Error: the value of id is not allowed to be undefined]
  details: 
   [ { message: 'the value of id is not allowed to be undefined',
       path: 'id',
       type: 'any.invalid' } ],
  _object: 
   { username: 'tcury',
     email: 'tcury@ymail.com',
     createdAt: Wed May 06 2015 12:15:09 GMT-0300 (Hora oficial do Brasil),
     updatedAt: Wed May 06 2015 12:15:09 GMT-0300 (Hora oficial do Brasil) },
  message: 'the value of id is not allowed to be undefined' }
C:\Users\tiago.cury\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\error\WLError.js:30
  this.rawStack = (this._e.stack.replace(/^Error(\r|\n)*(\r|\n)*/, ''));
                                ^
TypeError: Cannot read property 'replace' of undefined
    at new WLError (C:\Users\tiago.cury\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\error\WLError.js:30:33)
    at duckType (C:\Users\tiago.cury\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\error\index.js:67:10)
    at errorify (C:\Users\tiago.cury\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\error\index.js:39:10)
    at wrappedCallback (C:\Users\tiago.cury\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\utils\normalize.js:325:15)
    at callback.error (C:\Users\tiago.cury\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\node_modules\switchback\lib\normalize.js:42:31)
    at _switch (C:\Users\tiago.cury\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\node_modules\switchback\lib\factory.js:46:28)
    at c:\Users\tiago.cury\doceasy\node_modules\sails-dynamodb\index.js:848:11
    at c:\Users\tiago.cury\doceasy\node_modules\sails-dynamodb\node_modules\vogels\lib\table.js:119:14
    at start (c:\Users\tiago.cury\doceasy\node_modules\sails-dynamodb\node_modules\vogels\lib\table.js:108:12)
    at fn (c:\Users\tiago.cury\doceasy\node_modules\sails-dynamodb\node_modules\vogels\node_modules\async\lib\async.js:626:34)
    at Immediate._onImmediate (c:\Users\tiago.cury\doceasy\node_modules\sails-dynamodb\node_modules\vogels\node_modules\async\lib\async.js:542:34)
    at processImmediate [as _immediateCallback] (timers.js:358:17)
`

**Comment by devinivy** [comment]
I'll check it out.  I haven't tried creating a table using this adapter, though, and I'm not sure it currently has that ability.

**Comment by taddgiles** [comment]
I'm getting this same error, but I'm not trying to create a table, just a row. Here's the details:

node: 0.12.2  sails: 0.11.0   sails-dynamodb: 0.12.0

I have a model defined in Task.js with the following:

```js
module.exports = {
  attributes: {
    name: {
      type: 'string',
      required: true
    }
  }
};
```

I've created a dynamodb table name: **tasks** with a Hash Primary Key name: **id**, type: **string**

When I run the server locally connecting directly to the dynamodb server I run this curl command:

```bash
curl http://localhost:1337/tasks -v -X POST -H "Content-Type: application/json" -d '{ "name":"what" }'
```

The local server crashes with the following error:

```
/Users/Tadd/src/kuali/gitrdone/node_modules/sails/node_modules/waterline/lib/waterline/error/WLError.js:36
  this.rawStack = (this._e.stack.replace(/^Error(\r|\n)*(\r|\n)*/, ''));
                                ^
TypeError: Cannot read property 'replace' of undefined
    at new WLError (/Users/Tadd/src/kuali/gitrdone/node_modules/sails/node_modules/waterline/lib/waterline/error/WLError.js:36:33)
    at duckType (/Users/Tadd/src/kuali/gitrdone/node_modules/sails/node_modules/waterline/lib/waterline/error/index.js:67:10)
    at errorify (/Users/Tadd/src/kuali/gitrdone/node_modules/sails/node_modules/waterline/lib/waterline/error/index.js:39:10)
    at wrappedCallback (/Users/Tadd/src/kuali/gitrdone/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:320:15)
    at callback.error (/Users/Tadd/src/kuali/gitrdone/node_modules/sails/node_modules/waterline/node_modules/switchback/lib/normalize.js:42:31)
    at _switch (/Users/Tadd/src/kuali/gitrdone/node_modules/sails/node_modules/waterline/node_modules/switchback/lib/factory.js:46:28)
    at /Users/Tadd/src/kuali/gitrdone/node_modules/sails-dynamodb/index.js:848:11
    at /Users/Tadd/src/kuali/gitrdone/node_modules/sails-dynamodb/node_modules/vogels/lib/table.js:119:14
    at start (/Users/Tadd/src/kuali/gitrdone/node_modules/sails-dynamodb/node_modules/vogels/lib/table.js:108:12)
    at fn (/Users/Tadd/src/kuali/gitrdone/node_modules/sails-dynamodb/node_modules/vogels/node_modules/async/lib/async.js:626:34)
    at Immediate._onImmediate (/Users/Tadd/src/kuali/gitrdone/node_modules/sails-dynamodb/node_modules/vogels/node_modules/async/lib/async.js:542:34)
    at processImmediate [as _immediateCallback] (timers.js:358:17)

npm ERR! Darwin 14.3.0
npm ERR! argv "node" "/usr/local/bin/npm" "start"
npm ERR! node v0.12.2
npm ERR! npm  v2.7.6
npm ERR! code ELIFECYCLE
npm ERR! gitrdone@0.0.0 start: `sails lift`
npm ERR! Exit status 1
```

It should be auto generating the uuid for the id field, shouldn't it?

**Comment by taddgiles** [comment]
I worked around it by modifying my model to the following:

```js
var uuid = require('node-uuid');

module.exports = {
  autoPK: false,
  attributes: {
    id: {
      type: 'string',
      required: true,
      unique: true,
      primaryKey: true,
      defaultsTo: function () { return uuid.v4(); }
    },
    name: {
      type: 'string',
      required: true
    }
  }
};
```

But that shouldn't be necessary, should it?

**Comment by shinobiwarrior** [comment]
I change this line 
this.rawStack = (this._e.stack.replace(/^Error(\r|\n)*(\r|\n)*/, ''));
for this : 
this.rawStack = '';
in this archive WLError.js:36
and i send the id for each create method and work !!! 
until this issues is close.

**Comment by devinivy** [comment]
We need to fix table creation– moving this issue to #14.
